integrated_channel_plugin_sapsf/models.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals
+
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+from model_utils.models import TimeStampedModel, TimeFramedModel
+
+
+# Create your models here.
+
+class OCNWebServicesConfiguration(TimeStampedModel):
+    """
+    The Global configuration for integrating with SuccessFactors.
+    """
+
+    completion_status_api_path = models.CharField(max_length=255)
+    course_api_path = models.CharField(max_length=255)
+    oauth_api_path = models.CharField(max_length=255)
+    provider_id = models.CharField(max_length=100, default='EDX')
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+
+class OCNWebServicesEnterpriseCustomerConfiguration(TimeStampedModel):
+    """
+    The Enterprise specific configuration we need for integrating with SuccessFactors.
+    """
+
+    enterprise_customer_uuid = models.UUIDField(unique=True)
+    sapsf_base_url = models.CharField(max_length=255)
+    key = models.TextField(blank=True, verbose_name="Client ID")
+    secret = models.TextField(blank=True, verbose_name="Client Secret")
+
+    class Meta:
+        app_label = 'integrated_channel_plugin_sapsf'
+
+
+class CompletionStatusEventAudit(models.Model):
+    """
+    The payload we sent to SuccessFactors at a given point in time for an enterprise course enrollment.
+    """
+
+    enterprise_course_enrollment_id = models.PositiveIntegerField(blank=False, null=False)
+    sapsf_user_id = models.CharField(max_length=255, blank=False, null=False)
+    course_id = models.CharField(max_length=255, blank=False, null=False)
+    course_completed = models.BooleanField(default=True)
+    completed_timestamp = models.BigIntegerField()  # we send a UNIX timestamp to SuccessFactors
```

Comment:
Totally optional suggestion: if you want, you can remove the `course_completed` field and consider the course to be completed when `completed_timestamp` is non-null:

```python
@property
def course_completed(self):
    return self.completed_timestamp is not None
```